### PR TITLE
Cherry pick limit support

### PR DESCRIPTION
**Pull request by jbartosik**
Follow up to https://github.com/kubernetes/autoscaler/pull/2105 which had commits that cherry picked automatically. Some commits in this PR are not exact cherry pick and needed changes to

* vertical-pod-autoscaler/deploy/vpa-rbac.yaml
* vertical-pod-autoscaler/pkg/admission-controller/logic/server.go
* vertical-pod-autoscaler/pkg/admission-controller/logic/server_test.go
* vertical-pod-autoscaler/pkg/admission-controller/main.go

Because master has VPA preprocessor but vpa-release-0.5 doesn't (and some changes to RBAC). Each commit notes in its message any files I modified manually.

Cherry pick:
* https://github.com/kubernetes/autoscaler/pull/2049
* https://github.com/kubernetes/autoscaler/pull/2068
* https://github.com/kubernetes/autoscaler/pull/2072
* https://github.com/kubernetes/autoscaler/pull/2073
* https://github.com/kubernetes/autoscaler/pull/2076
* https://github.com/kubernetes/autoscaler/pull/2077
* https://github.com/kubernetes/autoscaler/pull/2079
* https://github.com/kubernetes/autoscaler/pull/2080
* https://github.com/kubernetes/autoscaler/pull/2091
* https://github.com/kubernetes/autoscaler/pull/2092
* https://github.com/kubernetes/autoscaler/pull/2093
* https://github.com/kubernetes/autoscaler/pull/2095
* https://github.com/kubernetes/autoscaler/pull/2099
* https://github.com/kubernetes/autoscaler/pull/2100

### CODE REVIEW

**Comment by jbartosik** [comment]
/assign @bskiba 
This had some merge conflicts but they were easy to resolve.

File: vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go [hunk 1]
```diff
@@ -183,6 +128,11 @@ func (p *recommendationProvider) GetContainersResourcesForPod(pod *v1.Pod) ([]Co
 			return nil, annotations, vpaConfig.Name, err
 		}
 	}
-	containerResources := GetContainersResources(pod, *recommendedPodResources, annotations)
+	podLimitRange, err := p.limitsRangeCalculator.GetContainerLimitRangeItem(pod.Namespace)
+	// TODO: Support limit range on pod level.
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go]: I just noticed this TODO is still here, is this expected?

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go]: No, I forgot to remove it (also variable name is misleading). I'll clean up in separate PRs.

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go]: https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go

**Comment by bskiba** [comment]
/lgtm
/approve 
Thanks!

**Pull request merged**